gp/primitives.py

```diff
@@ -26,33 +23,24 @@ def __init__(self, primitives: List[Type[Node]]) -> None:
             if not issubclass(primitives[i], Node):
                 raise TypeError(f"expected subclass of Node but received {primitives[i].__name__}")
 
-        self._primitives = {}
-        for i in range(len(primitives)):
-            self._primitives[i] = primitives[i]
-
-        # hide primitives dict behind MappingProxyType to make sure it
-        # is not changed after construction
-        # unfortunately not supported by pickle, necessary for
-        # multiprocessing; another way to implement this?
-        # self._primitives = types.MappingProxyType(self._primitives)
+        self._primitives = list(primitives)
+        self._max_arity: int = self._determine_max_arity()
```

Comment:
Wouldn't this be better written as a class property then?

Reply:
what is "this"? and why?

Reply:
`_max_arity`. This seems to me like the classic example where you want to use a class property: When accessing `_max_arity`, you want to call a routine that determines it on the fly. Or am I missing something?

Reply:
once initialized, this class is/should be treated as constant, the `max_arity` hence does not change afterwards. to save runtime it hence makes sense to compute it once during initialization